addons/binding/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/handler/SomfyTahomaBridgeHandler.java

```diff
@@ -0,0 +1,577 @@
+            @Override
+            public void run() {
+                try {
+                    updateTahomaStates();
+                } catch (Exception e) {
+                    logger.debug("Exception during poll : {}", e);
+                }
+            }
+        }, 10, refresh, TimeUnit.SECONDS);
+
+    }
+
+    private void login() {
+        String url = null;
+
+        if ((thingConfig.getEmail() != null && thingConfig.getEmail().isEmpty()) || (thingConfig.getPassword() != null && thingConfig.getPassword().isEmpty())) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "Can not access device as username and/or password are null");
+            return;
+        }
+
+        try {
+            url = TAHOMA_URL + "login";
+            String urlParameters = "userId=" + thingConfig.getEmail() + "&userPassword=" + thingConfig.getPassword();
+            byte[] postData = urlParameters.getBytes(StandardCharsets.UTF_8);
+
+            URL cookieUrl = new URL(url);
+            HttpsURLConnection connection = (HttpsURLConnection) cookieUrl.openConnection();
+            connection.setDoOutput(true);
+            connection.setRequestMethod("POST");
+            setConnectionDefaults(connection);
+            connection.setRequestProperty("Content-Length", Integer.toString(postData.length));
+            try (DataOutputStream wr = new DataOutputStream(connection.getOutputStream())) {
+                wr.write(postData);
+            }
+
+            //get cookie
+            String headerName;
+            for (int i = 1; (headerName = connection.getHeaderFieldKey(i)) != null; i++) {
+                if (headerName.equals("Set-Cookie")) {
+                    cookie = connection.getHeaderField(i);
+                    break;
+                }
+            }
+
+            InputStream response = connection.getInputStream();
+            String line = readResponse(response);
+
+            SomfyTahomaLoginResponse data = gson.fromJson(line, SomfyTahomaLoginResponse.class);
+
+            if (data.isSuccess()) {
+                String version = data.getVersion();
+                logger.debug("SomfyTahoma cookie: {}", cookie);
+                logger.info("SomfyTahoma version: {}", version);
+
+                for (Channel channel : getThing().getChannels()) {
+                    if (channel.getUID().getId().equals(VERSION)) {
+                        updateState(channel.getUID(), new StringType(version));
+                    }
+                }
+                updateStatus(ThingStatus.ONLINE);
+            } else {
+                logger.debug("Login response: {}", line);
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "Error logging in");
+                throw new SomfyTahomaException(line);
+            }
+        } catch (MalformedURLException e) {
+            logger.error("The URL '{}' is malformed: {}", url, e.toString());
```

Comment:
so you suggest exception shouldn't be logged by my code? logger.error("The URL '{}' is malformed", url); is sufficient in this case?